pkg/handler/v2/constraint/constraint.go

```diff
@@ -531,3 +531,41 @@ func DecodePatchConstraintReq(c context.Context, r *http.Request) (interface{},
 
 	return req, nil
 }
+
+func CreateDefaultEndpoint(userInfoGetter provider.UserInfoGetter,
+	defaultConstraintProvider provider.DefaultConstraintProvider,
+	constraintTemplateProvider provider.ConstraintTemplateProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createConstraintReq)
+
+		adminUserInfo, err := userInfoGetter(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+		if !adminUserInfo.IsAdmin {
+			return nil, utilerrors.New(http.StatusForbidden,
+				fmt.Sprintf("forbidden: \"%s\" doesn't have admin rights", adminUserInfo.Email))
+		}
+
+		constraint := convertAPIToInternalConstraint(req.Body.Name, ConstraintNamespace, req.Body.Spec)
```

Comment:
This ConstraintNamespace ("kubermatic") could be different from the namespace set for the Default Constraint Controller in the `seed-cluster-manager` . `kubermatic-api` also has the `namespace` option so maybe you can initialize the DefaultClusterProvider with it and use it through there somehow?

https://github.com/kubermatic/kubermatic/blob/master/cmd/kubermatic-api/options.go#L109

Reply:
done